segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -167,6 +172,33 @@ public void setupAppend(SetupAppend setupAppend) {
                 });
     }
 
+    @VisibleForTesting
+    void setupTokenExpiryTask(SetupAppend setupAppend, String newSegment, UUID writer) {
+        final Duration durationToExpiry = JwtUtils.durationToExpiry(setupAppend.getDelegationToken());
+
+        if (durationToExpiry != null) { // Can be null if token had no expiry set (for internal communications)
+            if (durationToExpiry.isNegative()) {
+                log.debug("Token has already expired");
```

Comment:
Include segment and writer id in the message. Both here and everywhere else you log or throw something.

Reply:
Sure, I've made this change.